### PR TITLE
CVSL-184: Add more additional condition child pages and form types

### PR DESCRIPTION
**Pull request by smcveigh941**
- Added `jest.setup.ts` to import `reflect-metadata` instead of needing to import it in every test
- In my next PR I intend to abstract the nunjucks filters into their own unit functions that can be more easily unit tested and mocked

### CODE REVIEW

File: integration_tests/pages/additionalConditionInput.ts [hunk 1]
```diff
@@ -26,6 +27,33 @@ export default class AdditionalConditionsInputPage extends Page {
     return this
   }
 
+  enterTime = (): AdditionalConditionsInputPage => {
+    cy.get("input[name*='hour']").type('11')
+    cy.get("input[name*='minute']").type('30')
```
Comment (timharrison-moj) [inline integration_tests/pages/additionalConditionInput.ts:32]: What does the star do on name* ?

Comment (timharrison-moj) [reply integration_tests/pages/additionalConditionInput.ts:32]: Just a wildcard match on name as it appears?

File: server/config/conditions.ts [hunk 2]
```diff
@@ -449,41 +533,106 @@ export default {
...
+          label: 'Enter the timeframe',
+          name: 'timeframe',
+        },
+        {
+          type: InputTypes.DATE,
+          name: 'endDate',
+        },
+      ],
+      type: ElectronicMonitoringPeriod,
+    },
```
Comment (timharrison-moj) [inline server/config/conditions.ts:605]: This condition has only one input field, but two values? Is this an either-or scenario? If a time-frame, the sentence won't make sense!

Comment (smcveigh941) [reply server/config/conditions.ts:605]: Oh! good spot, I hadn't realised the condition text had changed content. The inputs are correct, the text is not

File: integration_tests/pages/additionalConditionInput.ts (hunk 1 above)
Comment (smcveigh941) [reply integration_tests/pages/additionalConditionInput.ts:32]: This jQuery selector matches any input where the name attribute contains some text. For example we have inputs called `appointmentTime[hour]`  `inductionDate[hour]` `appointmentTime[minute]`  `inductionDate[minute]` etc. This selector matches any input with hour/minute as a field

Comment (smcveigh941) [reply integration_tests/pages/additionalConditionInput.ts:32]: https://api.jquery.com/attribute-contains-selector/

File: server/services/licenceService.ts [hunk 3]
```diff
@@ -159,12 +160,23 @@ export default class LicenceService {
...
+        if (Array.isArray(formData[key])) {
+          return formData[key].map((v: string, i: number) => {
+            return build(v, i)
+          })
+        }
+        if (formData[key] instanceof Stringable) {
+          return build(formData[key].stringify())
         }
+        return build(formData[key])
       }),
```
Comment (timharrison-moj) [inline server/services/licenceService.ts:180]: Perhaps worth a few comments around this part? e.g. Why some formData elements are arrays, others Stringable, others neither?

**Review by timharrison-moj — DISMISSED** [review]
LGTM. A bit hard to follow in one place, hence the comment suggestion there, but otherwise looks good.

File: server/services/licenceService.ts (hunk 3 above)
Comment (smcveigh941) [reply server/services/licenceService.ts:183]: Added a comment now and rejigged the logic a bit!

**Review by timharrison-moj — APPROVED** [review]
LGTM
